### ISSUE TITLE
pluginutils (via TypeScript plugin): Cannot find module 'estree'

### ISSUE THREAD

**Issue by meyfa** [rollup/plugins#1315]
- Rollup Plugin Name: `@rollup/plugin-typescript`
- Rollup Plugin Version: 9.0.1
- Rollup Version: 3.1.0
- Operating System (or Browser): Windows
- Node Version: 18.10.0
- Link to reproduction: https://github.com/meyfa/rollup-typescript-estree-repro

I'm getting an error with the newest version of the TypeScript plugin. This did not happen before the Rollup 3 preparations.

How to reproduce:

1. `npm install --save-dev typescript rollup @rollup/plugin-typescript`
2.  Define a config with the typescript plugin.
3. Perform a TypeScript typecheck, e.g. via `tsc --noEmit`.

### Expected Behavior

The typecheck succeeds.

### Actual Behavior

TypeScript outputs the following error:

```
node_modules/@rollup/pluginutils/types/index.d.ts:2:26 - error TS2307: Cannot find module 'estree' or its corresponding type declarations.

2 import { BaseNode } from 'estree';
                           ~~~~~~~~

Found 1 error in node_modules/@rollup/pluginutils/types/index.d.ts:2
```

### Additional Information

* `@rollup/plugin-typescript` still depends on `@rollup/pluginutils` version `^4.2.1`, while `5.0.0` would be available. Maybe this is part of the issue?
* I looked at some other issues with this error, especially #1156, but they were never resolved.

**Comment by meyfa** [comment]
```
> npm ls estree
rollup-typescript-estree-repro@0.0.0 C:\Users\meyfa\GitHub\rollup-typescript-estree-repro
`-- (empty)
```

Seems like estree isn't part of the dependency tree at all. On the other hand, estree-walker is. Maybe that one should be used? It also exports a `BaseNode` type.

**Comment by lukastaegert** [comment]
> Maybe that one should be used? It also exports a BaseNode type.

Good idea. I will see if I can prepare a PR that fixes this.

**Comment by lukastaegert** [comment]
Apparently, `entree-walker` is guilty of the same issue... Maybe we just need to put in that dependency. I wonder why this issue did not turn up for the previous version.

**Comment by meyfa** [comment]
Ah, sorry, I just realized we have to search the NPM tree for `@types/estree`, not `estree` itself. This gets us some useful information.

With `@rollup/plugin-typescript@8.5.0`:

```
> npm ls @types/estree
rollup-typescript-estree-repro@0.0.0 C:\Users\meyfa\GitHub\rollup-typescript-estree-repro
`-- @rollup/plugin-typescript@8.5.0
  `-- @rollup/pluginutils@3.1.0
    `-- @types/estree@0.0.39
```

With `@rollup/plugin-typescript@9.0.1`:

```
> npm ls @types/estree
rollup-typescript-estree-repro@0.0.0 C:\Users\meyfa\GitHub\rollup-typescript-estree-repro
`-- (empty)
```

This patch here seems guilty: https://github.com/rollup/plugins/commit/ce0652ddd9894c37ec47e1e04a291a77af5ce0ac

**Comment by meyfa** [comment]
I would argue that pluginutils should have `@types/estree` moved back into `dependencies`, since it does export something from there, which is clearly needed for downstream consumers of pluginutils.

**Comment by lukastaegert** [comment]
Agreed. Looking at the original PR, it looks like this change might have been an accident #616.

**Comment by meyfa** [comment]
I made a PR: #1320
